ddl/ddl_worker.go

```diff
@@ -426,11 +436,22 @@ func toTError(err error) *terror.Error {
 
 // waitSchemaChanged waits for the completion of updating all servers' schema. In order to make sure that happens,
 // we wait 2 * lease time.
-func (d *ddl) waitSchemaChanged(waitTime time.Duration) {
+func (d *ddl) waitSchemaChanged(waitTime time.Duration, latestSchemaVersion int64) {
 	if waitTime == 0 {
 		return
 	}
 
+	if ChangeOwnerInNewWay {
+		// TODO: Make ctx exits when the d is close.
+		ctx, cancelFunc := goctx.WithTimeout(goctx.Background(), waitTime)
+		defer cancelFunc()
```

Comment:
We only need to wait for 2*lease to make sure all servers' schema version are changed in the same cluster, If it's deadline, we needn't wait for it.